session-v7.go

```diff
@@ -258,7 +258,30 @@ func (s *sessionV7) Close() *probe.Error {
 	if err != nil {
 		return err.Trace(s.SessionID)
 	}
-	return qs.Save(sessionFile).Trace(sessionFile)
+
+	var currentHeader = &sessionV7Header{}
+	currentQS, err := quick.Load(sessionFile, currentHeader)
+	if err != nil {
+		// If session does not exist for the first, time attempt to
+		// create it no need to verify.
+		if os.IsNotExist(err.ToGoError()) {
+			return qs.Save(sessionFile).Trace(sessionFile)
+		}
+		return err.Trace(s.SessionID)
+	}
+
+	changedFields, err := qs.DeepDiff(currentQS)
+	if err != nil {
+		return err.Trace(s.SessionID)
+	}
+
+	// Save new if only values and fields have changed.
+	if len(changedFields) > 0 {
+		return qs.Save(sessionFile).Trace(sessionFile)
+	}
+
+	// Return success.
+	return nil
 }
```

Comment:
Can you move this into Session.save(), which in turn calls Session.isModified().